packages/core/test/signals/computed_spec.ts

```
@@ -0,0 +1,133 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {computed, signal} from '@angular/core/src/signals';
+
+describe('computed', () => {
+  it('should create computed', () => {
+    const counter = signal(0);
+
+    let computedRunCount = 0;
+    const double = computed(() => `${counter() * 2}:${++computedRunCount}`);
+
+    expect(double()).toEqual('0:1');
+
+    counter.set(1);
+    expect(double()).toEqual('2:2');
+    expect(double()).toEqual('2:2');
+
+    counter.set(2);
+    expect(double()).toEqual('4:3');
+    expect(double()).toEqual('4:3');
+  });
+
+  it('should not re-compute if there are no dependencies', () => {
+    let tick = 0;
+    const c = computed(() => ++tick);
```

Wondering if a computed with no dependencies should just throw an error since it's very likely not what the developer wants

---

Any side effects here should be avoided right? So this test looks like a misuse of `computed()`

---

> Wondering if a computed with no dependencies should just throw an error since it's very likely not what the developer wants

Computed might be used to wrap any arbitrary function call and sometimes it might be not clear if we expect signals read or not (ex. wrapping expressions provided to inputs).

---

> Any side effects here should be avoided right? So this test looks like a misuse of computed()

@kbrilla you are right that computed should be side-effect free but here we are explicitly testing re-computations.